spec/unit/puppet/provider/x509_request/openssl_spec.rb

```diff
@@ -22,26 +22,26 @@
     end
 
     it 'creates a certificate with the proper options' do
-      expect(provider_class).to receive(:openssl).with(
-        'req', '-new',
-        '-key', '/tmp/foo.key',
-        '-config', '/tmp/foo.cnf',
-        '-out', '/tmp/foo.csr'
-      )
+      expect(provider_class).to receive(:openssl).with([
+                                                         'req', '-new',
+                                                         '-key', '/tmp/foo.key',
+                                                         '-config', '/tmp/foo.cnf',
+                                                         '-out', '/tmp/foo.csr'
+                                                       ])
       resource.provider.create
     end
   end
 
   context 'when using password' do
     it 'creates a certificate with the proper options' do
       resource[:password] = '2x6${'
-      expect(provider_class).to receive(:openssl).with(
-        'req', '-new',
-        '-key', '/tmp/foo.key',
-        '-config', '/tmp/foo.cnf',
-        '-out', '/tmp/foo.csr',
-        '-passin', 'pass:2x6${'
-      )
+      expect(provider_class).to receive(:openssl).with([
+                                                         'req', '-new',
+                                                         '-key', '/tmp/foo.key',
+                                                         '-config', '/tmp/foo.cnf',
+                                                         '-out', '/tmp/foo.csr',
+                                                         ['-passin', 'pass:2x6${']
```

Comment:
… so this change the logic.  It seems not to hurt CI, but is it intended?

Reply:
The success of the tests depends on the method of passing the arguments to the openssl command. The endresult of what the class generates should not change, as the array is interpolated before the openssl command takes.
The tests however run before this interpolation, so if you change the method of passing arguments you also have to change the tests to expect an array instead of multiple items.